src/events/mod.rs

```diff
@@ -700,12 +700,25 @@ impl<'a> BytesCData<'a> {
     ///
     /// # Warning
     ///
-    /// `content` must not contain the `]]>` sequence.
+    /// `content` must not contain the `]]>` sequence. You can use
+    /// [`BytesCData::escaped`] to escape the content instead.
     #[inline]
     pub fn new<C: Into<Cow<'a, str>>>(content: C) -> Self {
         Self::wrap(str_cow_to_bytes(content), Decoder::utf8())
     }
 
+    /// Creates an iterator of `BytesCData` from a string.
```

Comment:
Could you move tests here as examples and also explain a bit why this method returns iterator (i. e. describe that we need to split user data at `]]>` to create well-formed XML)